src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,292 @@
+/**
+ * @file range_search_test.cpp
+ * @author Niteya Shah
+ *
+ * Test mlpackMain() of range_search_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "RangeSearchMain";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/range_search/range_search_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct RangeSearchTestFixture
+{
+ public:
+
+  RangeSearchTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+  ~RangeSearchTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(RangeSearchMainTest, RangeSearchTestFixture);
+/*
+* Check that the correct output is returned for a small synthetic input
+* case
+*/
+BOOST_AUTO_TEST_CASE(SyntheticRangeSearch)
+{
+  cout<<"Synthetic Test 1"<<endl;
+  //Matrix Input is expected in this format
+  arma::mat x={{0, 3, 3, 4, 3, 1},{4, 4, 4, 5, 5, 2},{0, 1, 2, 2, 3, 3}};
+  std::string distance_file="distances.csv";
+  std::string neighbors_file="neighbors.csv";
+  double min_v=0, max_v=3;
+  vector<std::vector<size_t>> neighbor_val={{},{2, 3, 4},{1, 3, 4, 5},{1, 2, 4},
```

Comment:
Why `vector<std::vector<size_t>>` and not the simpler `vector<vector<size_t>>`?  You're already `using namespace std` so it would make sense to make the type name shorter.